parsing/parsetree.mli

```diff
@@ -35,6 +35,7 @@ and constant_desc =
      Suffixes except ['l'], ['L'] and ['n'] are rejected by the typechecker
   *)
   | Pconst_char of char  (** Character such as ['c']. *)
+  | Pconst_uchar of Uchar.t  (** Unicode-only character such as ['𓆄']. *)
```

Comment:
It feels like the relation between Pconst_char and Pconst_uchar should be explained here. That is, Pconst_char holds constants of type either char or uchar, and the typer will decide which, whereas Pconst_uchar holds only constants of type uchar.

Then a followup question would be: is it an invariant that Pconst_uchar does not contain plain chars? I doubt it (some of the typer code is using untypeast which should create things like Pconst_uchar 'a'), in which case the documentation is misleading and it should merely claim that the parser only creates this case with code points >= 128.